### ISSUE TITLE
[Serve] Error status message isn't cleared after service redeployed correctly

### ISSUE THREAD

**Issue by galenhwang** [ray-project/ray#35508]
### What happened + What you expected to happen

Status message isn't cleared after service goes from `DEPLOY_FAILED` to `RUNNING`

### Versions / Dependencies

master (commit 658c8e3)

### Reproduction script

1. Call `serve deploy` with incorrect runtime_env
```
runtime_env:
    working_dir: "https://not_a_valid_website.com/HEAD.zip"
```

![Screenshot 2023-05-18 at 11 32 46 AM](https://github.com/ray-project/ray/assets/29215801/5a7c9e37-b30b-4882-aa5e-1493966cca37)

2. Call `serve deploy` with correct runtime_env, however error status message remains

```
    working_dir: "https://github.com/ray-project/serve_config_examples/archive/HEAD.zip"
```

![Screenshot 2023-05-18 at 11 33 10 AM](https://github.com/ray-project/ray/assets/29215801/7bac9741-3d76-43ee-8958-44b05f193a32)

### Issue Severity

None

**Comment by scottsun94** [comment]
I ran into the same issue.

**Comment by scottsun94** [comment]
@edoakes can we prioritize this fix and cherrypick into 2.5 if possible?

**Comment by edoakes** [comment]
@shrekris-anyscale will prioritize and we should cherry-pick (assuming fix is very low risk, which it should be)
